manipmongo/utils.go

```diff
@@ -278,11 +279,12 @@ func spanErr(sp opentracing.Span, err error) error {
 // HandleQueryError handles the provided upstream error returned by Mongo by returning a corresponding manipulate error type.
 func HandleQueryError(err error) error {
 
-	if _, ok := err.(net.Error); ok {
+	var netErr net.Error
+	if errors.As(err, &netErr) {
```

Comment:
do we need errors.As ? we don't use the netErr instance it seems

Reply:
`net.Error` is an interface which actually makes it more annoying as we cannot just create an empty object to compare against and instead have to declare it on a separate line first. `Is` won't work here since `net.Error` is not of type `error`